### PR TITLE
Update WebRTC setup guide

### PR DESCRIPTION
**Pull request by jasonblais**
@enahum left a few "XXX" notes, can you help review those?

Please also review the doc for accuracy, based in on your notes: https://docs.google.com/document/d/142tZMQpOSSUMN19fFQ9c1n4mh4_w5B9UC1Ek3bhWCtI/edit#heading=h.9lpl2o1xbd09

### CODE REVIEW

File: source/deployment/webrtc.rst [hunk 1]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
 
 .. note:: This feature will be replaced by a plug-in architecture allowing multiple video and audio calling providers to connect into Mattermost, and features described in this section will be re-written as a plug-in.
 
-Configuring video and audio calls
-------------------------------------------
+Configuring video and audio calls with WebRTC
+----------------------------------------------
 
-This option can be enabled by the System Administrator in the System Console under **Integrations > WebRTC (Beta)** - `see configuration settings documentation to learn more <https://docs.mattermost.com/administration/config-settings.html#webrtc-beta>`_.
+This guide is aimed to set up Mattermost WebRTC with a Docker image, `available here <https://hub.docker.com/r/mattermost/webrtc/>`_  and a Janus server to act as the WebRTC gateway.
```
Comment (enahum) [inline source/deployment/webrtc.rst]: The docker image is the janus gateway server

File: source/deployment/webrtc.rst [hunk 2]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
...
-Configuring video and audio calls
-------------------------------------------
+Configuring video and audio calls with WebRTC
+----------------------------------------------
 
-This option can be enabled by the System Administrator in the System Console under **Integrations > WebRTC (Beta)** - `see configuration settings documentation to learn more <https://docs.mattermost.com/administration/config-settings.html#webrtc-beta>`_.
+This guide is aimed to set up Mattermost WebRTC with a Docker image, `available here <https://hub.docker.com/r/mattermost/webrtc/>`_  and a Janus server to act as the WebRTC gateway. 
 
-To set up the WebRTC server, you may either
+If you want to use a full `Janus Gateway <https://janus.conf.meetecho.com/>`_, please visit their `GitHub repo <https://github.com/meetecho/janus-gateway>`_ for detailed instructions. You may also optionally set up `Coturn <https://github.com/coturn/coturn/wiki>`_ for STUN and TURN servers for your Mattermost installation.
```
Comment (enahum) [inline source/deployment/webrtc.rst]: Coturn is just a Turn server if im right

File: source/deployment/webrtc.rst [hunk 3]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
...
 
- - use a `Mattermost Docker container created for testing WebRTC <https://hub.docker.com/r/mattermost/webrtc/>`_
- - set up a `Janus server <https://github.com/meetecho/janus-gateway>`_ to act as the WebRTC gateway and `Coturn <https://github.com/coturn/coturn/wiki>`_ for STUN and TURN servers for your Mattermost installation.
+Pre-requisites
+~~~~~~~~~~~~~~~
 
-Starting a video call
+- Install Docker using the `Ubuntu online guide <https://docs.docker.com/installation/ubuntulinux/`_. Docker installation, configuration and management are out of scope for this guide.
+- Install a Mattermost server using the `official install guides <https://docs.mattermost.com/guides/administrator.html#installing-mattermost>`_. Mattermost server installation, configuration and management are out of scope for this guide.
+- Install Janus server version 0.2.2.
```
Comment (enahum) [inline source/deployment/webrtc.rst]: The Janus server as I said is part of the mattermoat webrtc docker container

File: source/deployment/webrtc.rst [hunk 4]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
...
+- Add an SSL certificate for host **dockerhost**, valid until January 2, 2018.
```
Comment (enahum) [inline source/deployment/webrtc.rst]: Also included with the container

File: source/deployment/webrtc.rst [hunk 5]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
...
+- Ability to connect using SSL or plain WebSocket and HTTP // XXX Is this for WebRTC connection?
+
+STUN and TURN servers are not required for this setup.
```
Comment (enahum) [inline source/deployment/webrtc.rst]: Incorrect you need at least one stun server that is why I recommended the google public one

File: source/deployment/webrtc.rst [hunk 6]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
...
+After installing Docker, run the following command in a terminal to install the Mattermost WebRTC Docker image.
+
+   .. code:: bash
+
+       docker run --name mattermost-webrtc -p 7088:7088 -p 7089:7089 -p 8188:8188 -p 8189:8189 -d mattermost/webrtc:latest
+
+The command downloads, installs and runs your ``mattermost-webrtc`` container with the Janus Gateway pre-configured to use WebRTC on Chrome, Firefox or the Mattermost Destkop Apps.
+
+.. note::
+  Make sure your Mattermost server can reach the running Mattermost WebRTC Docker container. // XXX How can they verify this?
```
Comment (enahum) [inline source/deployment/webrtc.rst]: There are different ways to verify this but one way is issuing a telnet to one of the ports listed above

File: source/deployment/webrtc.rst [hunk 7]
```diff
@@ -7,17 +7,65 @@ The feature is a working prototype for community development and not recommended
...
+
+Configure Mattermost to enable WebRTC
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1 - (Optional) Only required if you want to establish the connection to the Mattermost WebRTC Docker container, running the Janus Gateway, with SSL.
+
+Mattermost makes an HTTP request to the Janus Gateway service to get a ``Token``, which is used to identify a user and establish a connection between peers. If you configure Mattermost to make requests to that service with SSL, the SSL certificate used to run the Janus Gateway will not be considered signed by a trusted CA.
```
Comment (enahum) [inline source/deployment/webrtc.rst]: Make sure that is understood that this means the SSL certificate included in the docker image

File: source/deployment/webrtc.rst (hunk 4 above)
Comment (jasonblais) [inline source/deployment/webrtc.rst]: Right that makes sense :)

**Comment by jasonblais** [comment]
Thanks for the review @enahum! Have updated based on feedback.

File: source/deployment/webrtc.rst [hunk 8]
```diff
@@ -7,17 +7,66 @@ The feature is a working prototype for community development and not recommended
...
+
+.. image:: ../images/webrtc_full_settings.png
+
+- **Enable Mattermost WebRTC** - ``true``.
+- **Gateway WebSocket URL** - example: ``wss://dockerhost:8189``. WebSocket route for the Janus Gateway service, inside the Mattermost WebRTC container, used to connect peers on a video call. For SSL connections, set the protocol to ``wss://`` and the port to ``8189``. For non-SSL connections, set the protocol to ``ws://`` and port to ``8188``.
+- **Gateway Admin URL** - example: ``https://dockerhost:7089/admin``.  Admin route for the Janus Gateway service inside the Mattermost WebRTC container, used to fetch a valid ``Token``. For SSL connections, set the protocol to ``https://`` and the port to ``7089``. For non-SSL connections, set the protocol to ``http://`` and port to ``7088``.
+- **Gateway Admin Secret** - example: ``janusoverload``: The secret that validates the request made to fetch the ``Token``. For any Janus Gateway installation the default value is ``janusoverlord``. Change it by editing the ``janus.cfg`` file under the ``/opt/janus/etc/janus`` directory, and modifying the value for ``admin_secret``.
```
Comment (lindy65) [inline source/deployment/webrtc.rst]: Should `janusoverlord` be `janusoverload`?

**Comment by enahum** [comment]
Nope is janusoverlord

**Comment by esethna** [comment]
Looks good, made a few minor tweaks

**Pull request merged**